src/main/java/domen/rideapp/infrastructure/pricing/CustomPricingService.java

```diff
@@ -0,0 +1,38 @@
+package domen.rideapp.infrastructure.pricing;
+
+import domen.rideapp.domain.model.Localization;
+import domen.rideapp.domain.model.PricingConfig;
+import domen.rideapp.domain.model.RouteEstimate;
+import domen.rideapp.domain.repository.PricingRepository;
+import domen.rideapp.domain.service.MapService;
+import domen.rideapp.domain.service.PricingService;
+
+public class CustomPricingService implements PricingService {
+    private final MapService mapService;
+    private final PricingRepository pricingRepository;
+
+    public CustomPricingService(MapService mapService, PricingRepository pricingRepository) {
+        this.mapService = mapService;
+        this.pricingRepository = pricingRepository;
+    }
+
+    private RouteEstimate getEstimateOrThrow(Localization localization) {
```

Comment:
Jeśli jest wykorzystywana tylko w jednym miejscu to ja bym nie tworzył prywatnej metody tylko wywołał kod bezpośrednio w metodzie publicznej zwłaszcza że nie jest ona skomplikowana, ale decyzje pozostawiam Tobie